docs/MachineConfiguration.md

```diff
@@ -180,6 +181,29 @@ spec:
 
 **Note:** The RT kernel lowers throughput (performance) in return for improved worst-case latency bounds. This feature is intended only for use cases that require consistent low latency. For more information, see the [Linux Foundation wiki](https://wiki.linuxfoundation.org/realtime/start) and the [RHEL RT portal](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux_for_real_time/8/).
 
+### RHCOS Extensions
+RHCOS is a minimal OCP focused OS which provides capabilities common across all the platforms. With extensions support, OCP 4.6 and onward users can enable a limited set of additional functionality on the RHCOS nodes. In OCP 4.6 the supported extensions are `usbguard` and `kernel-devel`.
+
+**Note:** `kernel-devel` extension should be installed only when you are using compatible traditional kernel (kernleType: default). If a user tries to install on an incompatible kernel (e.g. kernelType: realtime), MachineConfig will not be rendered and relevant errors can be seen in the machine-config-controller pod.
```

Comment:
Let's do docs PRs in a hackmd.io or something to start?  There's e.g. a `kernleType` typo and other things that would be simpler to just fix and not do a review/repush cycle.